include/qinterface.hpp

```diff
@@ -368,6 +368,61 @@ class QInterface {
     virtual void ApplyAntiControlledSingleBit(
         const bitLenInt* controls, const bitLenInt& controlLen, const bitLenInt& target, const complex* mtrx) = 0;
 
+    /**
+     * Apply a single bit transformation that only effects phase.
+     *
+     * If float rounding from the application of the matrix might change the state vector norm, "doCalcNorm" should be
```

Comment:
Always a little concerned by statements like this which imply that certain parameters are dependent on other parameters.  Without having looked further in the PR - is this something that could be internalized?  Or, perhaps, more actionable guidance provided?

Reply:
The point of separating these out, is that they are cases which can gain very significant optimization. By adding these methods to the interface, we avoid having to overload every individual basic gate in QUnit and other derived classes, to achieve something similar but not even as good.

Reply:
Addtionally, while there's no problem with making these additions protected, exposing them publicly lets users directly access the optimized cases, rather than only being able to access them through the basic wrapper gates. So, I think they're worth making public.